README.md

```diff
@@ -1,23 +1,147 @@
 # :flamingo: flatzingo
 
-A flatzinc frontend for clingcon
+A [FlatZinc](https://www.minizinc.org/doc-2.4.3/en/flattening.html) frontend to solve CP problems in [MiniZinc](https://www.minizinc.org) format using [clingcon](https://potassco.org/clingcon/) as solver.
 
-Basically the following pipeline:
+The process is done in two stages:
+1. Transform files from MiniZinc in the flattened representation FlatZinc using [fzn2lp](https://github.com/potassco/fzn2lp) into `lp` files.
+2. With the encodings in files [encoding](encoding.lp) and [types](types.lp), the `lp` file is proceeded to use the constraints with the syntax of `clingcon`.
 
-For checking of unsupported constraints,
-``` sh
-{minizinc -c -G ../../../share/minizinc/flatzingo --output-fzn-to-stdout example.mzn | fzn2lp; cat static_check.lp types.lp }| clingcon
+
+## Install
+
+### Using Docker
+
+Follow the steps in the last [section](install-using-a-docker-image). *Note that **Usage** will vary on the paths.*
+
+### Manual Installation
+
+1. Install MiniZinc as showed [here](https://www.minizinc.org/doc-2.5.5/en/installation.html)
+2. MiniZinc requires the [files](share/minizinc/flatzingo) defining the high-level constraints used by flatzingo, to be on a relative path to the installation of MiniZinc. So they have to be copied to these directory. Given the MiniZinc path added during the instalation `$PATH_MINIZINC` copy the files with: 
+```
+cp share/minizinc/flatzingo/* $PATH_MINIZIC/share/minizinc/flatzingo
+```
+Example in MacOS
+```
+cp share/minizinc/flatzingo/* /Applications/MiniZincIDE.app/Contents/Resources/share/minizinc/flatzingo
+```
+3. [Install fzn2lp](https://github.com/potassco/fzn2lp) manually using `cargo`
+4. [Install clingcon](https://github.com/potassco/clingcon) 
+
+## Usage
+
+Run MiniZinc (Without solving `-c`) to get a FlatZinc output (`--output-fzn-to-stdout`) from a MiniZinc model (in this case `encodings/example.mzn`). Additionally the path to the high-level constraints (moved during the installation) must be specified with the `-G` option.
+The output is then piped to `fzn2lp`
+
+```
+minizinc  -G -c flatzingo --output-fzn-to-stdout encodings/example.mzn | fzn2lp > tmp.lp
+```
+
+Use the `lp` file obtained in a `clingcon` call with other encodings. 
+
+- **Static check**
+  
+Check the problem before converting it to clingcon's syntax
+```
+clingcon static_check.lp types.lp tmp.lp
 ```
 
-computing a solution,
-``` sh
-{minizinc -c -G ../../../share/minizinc/flatzingo --output-fzn-to-stdout example.mzn | fzn2lp; cat encoding.lp types.lp }| clingcon
+- **Compute solution with `clingcon`**
+
+```
+clingcon encodings/encoding.lp encodings/types.lp tmp.lp
```

Comment:
output missing

Reply:
Do you mean I should redirect this as well? What type of file would you like in this case? just txt?